src/plugins/email/email.c

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file
+ *
+ * @brief Source for email plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdberrors.h>
+#include <regex.h>
+#include <stdio.h>
+
+#include "email.h"
+
+static int validateEmail (const char * addr)
+{
+	if (!addr) return 0;
+	const char * regexString = "^[a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~-]([a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~.-]{0,61}[a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~-])?@([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)(\\.[a-zA-Z0-9]([a-zA-Z0-9-.]{0,61}[a-zA-Z0-9])?)+$";
```

Comment:
How did you come up with the regex or where does it come from?

Reply:
Regex is based on syntax part of wikipedia article https://en.wikipedia.org/wiki/Email_address.
I thought it to be a good summary of the RFCs